Components/ProductFromShop.php

```diff
@@ -500,6 +481,39 @@ public function calculateShippingCosts(Order $order)
             Shopware()->Modules()->Basket()->sAddArticle($product['ordernumber'], $product['quantity']);
         }
 
+        $shippingMethods = Shopware()->Modules()->Admin()->sGetPremiumDispatches($country->getId());
+        if (empty($shippingMethods)) {
+            return new Shipping([
+                'isShippable' => false,
+                'messages' => [
+                    new Message([
+                        'message' => 'order_not_shippable_to_country',
+                        'values' => [
+                            'country' => $countryIso3,
+                        ]
+                    ])
+                ]
+            ]);
+        }
+
+        $shippingMethod = reset($shippingMethods);
```

Comment:
how are these shippingcosts ordered? because we just take the first but doesn't it make sense to take the cheapest?

Reply:
Shopware will decide which shipping cost method to return by given country and products in basket. You can see in ```Shopware_Controllers_Frontend_Checkout::getSelectedDispatch```

```
/**
     * Get selected dispatch or select a default dispatch
     *
     * @return bool|array
     */
    public function getSelectedDispatch()
    {
        if (empty($this->session['sCountry'])) {
            return false;
        }

        $dispatches = $this->admin->sGetPremiumDispatches($this->session['sCountry'], null, $this->session['sState']);
        if (empty($dispatches)) {
            unset($this->session['sDispatch']);

            return false;
        }

        foreach ($dispatches as $dispatch) {
            if ($dispatch['id'] == $this->session['sDispatch']) {
                return $dispatch;
            }
        }
        $dispatch = reset($dispatches);
        $this->session['sDispatch'] = (int) $dispatch['id'];

        return $dispatch;
    }
```